src/controls/Controls.spec.js

```diff
@@ -1 +1,62 @@
 // Test away!
+import React from "react";
+import { render, fireEvent, cleanup } from "react-testing-library";
+
+import Controls from "./Controls";
+
+afterEach(() => cleanup());
+
+describe("The Controls Component", () => {
+  it("provides button to toggle locked state", () => {
+    const mockLock = jest.fn();
+    const { getByText } = render(<Controls closed toggleLocked={mockLock} />);
+    const lockButton = getByText(/lock gate/i);
+    fireEvent.click(lockButton);
+    expect(mockLock).toHaveBeenCalled();
+  });
+  it("provides button to toggle closed state", () => {
+    const mockClose = jest.fn();
+    const { getByText } = render(<Controls toggleClosed={mockClose} />);
+    const closeButton = getByText(/close gate/i);
+    fireEvent.click(closeButton);
+    expect(mockClose).toHaveBeenCalled();
+  });
+  describe("changes button text to reflect state", () => {
+    it("displays unlock button when locked prop is passed", () => {
+      const { getByText } = render(<Controls locked />);
+      getByText(/unlock gate/i);
+    });
+    it("displays lock button when unlocked", () => {
+      const { getByText } = render(<Controls />);
+      getByText(/lock gate/i);
+    });
+    it("displays close button when open", () => {
+      const { getByText } = render(<Controls />);
+      getByText(/close gate/i);
+    });
+    it("displays open button when closed", () => {
+      const { getByText } = render(<Controls closed />);
+      getByText(/open gate/i);
+    });
+  });
+  describe("buttons disable depending on locked and closed status", () => {
+    it("closed toggle button is disabled if gate is locked and closed", () => {
+      const disabledToggle = jest.fn();
+      const { getByText } = render(
+        <Controls locked closed toggleClosed={disabledToggle} />
+      );
+      const button = getByText(/open gate/i);
+      fireEvent.click(button);
+      expect(disabledToggle).toHaveBeenCalledTimes(0);
+      cleanup();
+    });
+    it('locked toggle button is disabled when door is open', () => {
+      const disabledToggle = jest.fn();
+      const { getByText } = render(<Controls />)
+      const button = getByText(/lock gate/i);
+      fireEvent.click(button);
+      expect(disabledToggle).toHaveBeenCalledTimes(0);
```

Comment:
You've used a similar pattern to the one in this test function already; do you think you could combine the logic into a helper function? Something like this is what comes to my mind:
```javascript
const makeControlsTester = (props, { toggle, text, times = 1 }) => {
  return () => {
    const mockFn = jest.fn();
    if (toggle) props[toggle] = mockFn;
    const { getByText } = render(
      <Controls  {...props } />
    );
    const button = getByText(text);
    expect(mockFn).toHaveBeenCalledTimes(times);
    // cleanup is already performed after each test
  };
};
```
and, the usage for the first test and the last test:
```javascript
// first test
it("provides button to toggle locked state",
  makeControlsTester({ 
    closed: true 
  }, {
    toggle: 'toggleLocked',
    text: /lock gate/i
  })
);

// ...

// last test
it("locked toggle button is disabled when door is open",
  makeControlsTester({}, {
    text: /lock gate/i,
    times: 0
  })
);
```